### PR TITLE
[close #220] TiKV-BR support key range

### LINKED ISSUE #220: TiKV-BR Support keyrange in backup&restore
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**
TiKV-BR Support keyrange in backup&restore
tikv-br backup raw --pd={pd} -s={storage} -start="a" --end="z" --format="hex"

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by haojinming**
### What problem does this PR solve?

Issue Number: close #220 

Problem Description: 

### What is changed and how does it work?
Support key range in backup & restore.

### Code changes

- Has exported function/method change

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Side effects

- No side effects

### Related changes

- No related changes

### CODE REVIEW

**Review by zeminzhou — APPROVED** [review]
LGTM~

File: br/tests/rawkv/main.go [hunk 1]
```diff
@@ -222,82 +251,100 @@ func CheckBackupTS(apiVersion kvrpcpb.APIVersion, tso uint64, backupOutput []byt
...
 
-	if apiVersion == kvrpcpb.APIVersion_V1TTL {
+	prefix := []byte("index")
+	q1Key, _ := GenerateTestData(*keyCnt/4, prefix)
+	q3Key, _ := GenerateTestData(3**keyCnt/4, prefix)
+	backupRanges := []kvrpcpb.KeyRange{
+		{StartKey: []byte{}, EndKey: []byte{}},
+		{StartKey: q1Key, EndKey: q3Key},
+	}
+	for _, keyRange := range backupRanges {
```
Comment (pingyu) [inline br/tests/rawkv/main.go:321]: Nit: it seems better to treat `backupRanges` as test cases, and move the loop to `main`.

File: br/tests/rawkv/main.go [hunk 2]
```diff
@@ -222,82 +251,100 @@ func CheckBackupTS(apiVersion kvrpcpb.APIVersion, tso uint64, backupOutput []byt
...
+			log.Panic("Preload data fail", zap.Error(err))
+		}
+		runBackupAndRestore(ctx, tester, prefix, keyRange.StartKey, keyRange.EndKey)
+
+		if apiVersion == kvrpcpb.APIVersion_V1TTL {
+			if err := tester.ClearStorage(); err != nil {
+				log.Panic("ClearStorage fail", zap.Error(err))
+			}
+			safeInterval := int64(120) // 2m
+			backupOutput, err := tester.Backup(ctx, kvrpcpb.APIVersion_V2, safeInterval, keyRange.StartKey, keyRange.EndKey)
```
Comment (pingyu) [inline br/tests/rawkv/main.go:340]: Suggest to also verify the backup result by restore + checksum.

File: br/pkg/gluetikv/glue_test.go [hunk 3]
```diff
@@ -15,12 +15,24 @@
...
 )
 
-func TestGetVersion(t *testing.T) {
+func TestGlue(t *testing.T) {
 	g := Glue{}
 	require.Regexp(t, "^BR(.|\n)*Release Version(.|\n)*Git Commit Hash(.|\n)*$", g.GetVersion())
+
+	require.True(t, g.OwnsStorage())
+
+	ctx := context.Background()
```
[the remark this answers is not in the document]
Comment (haojinming) [reply br/pkg/gluetikv/glue_test.go:30]: Just to cover the code to verify no crash occurs.

File: br/tests/rawkv/main.go (hunk 2 above)
Comment (haojinming) [reply br/tests/rawkv/main.go:340]: It's a bit problem here, we only deploy one cluster in integration test. May need deploy apiv1 & apiv2 clusters to verify the conversion result.

File: br/tests/rawkv/main.go (hunk 1 above)
Comment (haojinming) [reply br/tests/rawkv/main.go]: Done

File: br/tests/rawkv/main.go (hunk 2 above)
Comment (pingyu) [reply br/tests/rawkv/main.go]: Got it. We may consider to deploy two clusters for this test case later.

**Review by pingyu — APPROVED** [review]
LGTM~

File: br/tests/rawkv/main.go (hunk 2 above)
Comment (haojinming) [reply br/tests/rawkv/main.go]: Ok，I will improve it later.

**Pull request merged**